src/scores/continuous/standard_impl.py

```
@@ -561,3 +563,208 @@ def kge(
+        bound will no longer be 1.0. (e.g. when the forecast error is positive, but the obs variance
+        is negative, and vice-versa.)
+        """ 
+        scores.utils.check_weights_positive(weights, context=context)
+        self.weights = weights
+        return self
+
+    def _with_additional_reduce_dims(self: ..., reduce_dims: DimNameCollection) -> NseBuilder:
+        # implicitly type checks dimension names
+        self.reduce_dims = scores.utils.merge_dim_names(self.time_dim, reduce_dims)
+        return self
+
+    def _with_angular(self: ..., is_angular: bool) -> NseBuilder:
+        # automatically handled by underlying mse function
+        self.is_angular = is_angular
+        return self
+    # ---
+
+    # nse_impl.Nse_builder.build(...) - but actually can also be "post init"
+    def _build():
+        _mse: Callback[..., XarrayLike] = self._build_mse_callback()
+        fcst_error: XarrayLike = _mse(fcst, obs)
+        obs_variance: XarrayLike = _mse(obs_mean, obs)
+
+        # ---
+        # TODO: move to separate functions
+        # check decomposed scores...
+        # NOTE: these checks will automatically trigger data loads since conditionals cannot be used
+        # with dask delayed. but they have to be checked for safety.
+        if obs_variance == 0:
+            warnings.warn("Divide by zero...", UserWarning)
+        # ---
+
+        return NseScore(fcst_error)
+
+        # optional - self.destroy():
+        # NOT implemented, because its hard to know what data needs to be kept if dask is involved.
+
+
+    # ---------------------------------------------------------------------------------------------
+    # Score impl - move to nse_impl.NseScore.nse
+    # ---------------------------------------------------------------------------------------------
+    def _nse(self: ..., *) -> XarrayLike:
+        # warning should already have been issued by builder
+        with np.errstate(divide="ignore"):
```

comment: ignores errors and warnings to do with divide by 0. `np.nan` is the result for each element that divide by zero happens.